state-chain/chains/src/sol/instruction_builder.rs

```diff
@@ -72,7 +75,25 @@ impl SolanaInstructionBuilder {
 
 		final_instructions.append(&mut instructions);
 
-		final_instructions
+		// Test serialize the final transaction to obtain its length.
+		let transaction = SolTransaction::new_unsigned(SolMessage::new_with_blockhash(
+			&final_instructions,
+			Some(&agg_key),
+			&durable_nonce.1.into(),
```

Comment:
You have changed the `nonce_account: SolPubkey` to `durable_nonce: DurableNonceAndAccount` in all these functions to then use that durable nonce here when mocking the final serialized transaction. I think that's unnecessary. We are already crafting this mocked transaction with an empty signature (using `new_unsigned` that uses `SolSignature::default()`) so we could do the same with the blockhash. Then we don't need to do those changes and pass the durable nonce value.

Reply:
you could either just add the default hash value or add the `new` function to the `Message`:
```
pub fn new(instructions: &[Instruction], payer: Option<&Pubkey>) -> Self {
Self::new_with_blockhash(instructions, payer, &Hash::default())
}
```

Reply:
yeah you are right, using the default hash has no affect on the length of the final transaction.